src/com/amazon/corretto/crypto/provider/EvpKemKey.java

```diff
@@ -0,0 +1,42 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.corretto.crypto.provider;
+
+public abstract class EvpKemKey extends EvpKey {
+
+  private volatile Integer parameterSet;
```

Comment:
I think a lot of places where you are using `Integer parameterSet` might be better served by a new `MlkemParameter` Java Enum with values of `MLKEM_512`, `MLKEM_768`, and `MLKEM_1024`. 

Something along the lines of:
```
enum MlkemParameter {
   MLKEM_512(...),
   MLKEM_768(...),
   MLKEM_1024(...);
   
   static MlkemParameter fromParameterSize(int parameterSet) { ... }
   
   int getPublicKeySize() { ... }
   int getSecretKeySize() { ... }
   int getCiphertextSize() { ... }
   int getParameterSize() { ... }
}
```